indico_toolkit/staggered_loop/staggered_loop.py

```diff
@@ -20,11 +43,484 @@ def __init__(self, client: IndicoClient):
+                labelsets{
+                    id
+                    name
+                }
+            }
+        }
+        """
+        response = self.graphQL_request(
+            graphql_query=query, variables={"id": dataset_id}
+        )
+        dataset_details = response.get("dataset")
+        if not dataset_details:
+            raise ValueError(f"No dataset found for id={dataset_id}")
+        print(f"Fetched dataset metadata dataset_id={dataset_id}")
+
+        return dataset_details
+
+    def get_review_data(
+        self, workflow_id: int, update_date: datetime
+    ) -> List[WorkflowResult]:
+        """
+        Fetch review data from workflow, using the update_date as a filter
+
+        TODO Could be preferable to filter data by completedAt timestamp instead
+         of updatedAt, but that is more complex
+
+        Will also run pre review and post review comparisons here and
+        add TP/FP/FN metadata as well
+
+        Args:
+            workflow_id: ID of production workflow with review enabled
+            update_date: Update date for dataset in dev. Used to only capture
+                submissions that have been updated since then
+
+        Returns:
+            workflow_results: Review data fetched from workflow
+        """
+        # Get list of submission IDs using GetDocumentReport query
+        submission_ids = []
+        document_filter = DocumentReportFilter(
+            # TODO Comment back in when latest version of the client has been deployed
+            #  Earlier version of client does not have status as an argument to filter
+            # status="COMPLETED",
+            workflow_id=workflow_id,
+            updated_at_start_date=update_date,
+            updated_at_end_date=datetime.datetime.now(),
+        )
+        for page in self.client.paginate(GetDocumentReport(filters=document_filter)):
+            for submissions in page:
+                submission_ids.extend(
+                    [sub.submission_id for sub in submissions.input_files]
+                )
+
+        # Fetch results from submission IDs
+        self._workflow_results = self.get_submission_results_from_ids(
+            submission_ids=submission_ids,
+            ignore_deleted_submissions=True,
+        )
+        print(
+            f"Downloaded review data from workflow_id={workflow_id} that was updated since"
+            f"date={update_date} num_submissions={len(self._workflow_results)}"
+        )
+        return self._workflow_results
+
+    def get_submission_full_text(self):
+        """
+        Fetch full text (OCR) from all workflow submissions. Should be run after
+        get_review_data()
+        """
+        for wf_result in self._workflow_results:
+            ondoc_result = self.get_ondoc_ocr_from_etl_url(wf_result.etl_url)
+            self._document_texts.append(ondoc_result.full_text)
+        print("Downloaded text (OCR) for all workflow submissions")
+
+    def get_document_bytes(self):
```

Comment:
Seems like this still needs a rework.

Reply:
Still think we would want the change to pull the input file rather than making a PDF from page images (which will absolutely mess up offsets).  The text comparison is crucial at least for testing.